taichi/transforms/ir_printer.cpp

```diff
@@ -805,9 +805,9 @@ void print(IRNode *root, std::string *output) {
   return IRPrinter::run(&expr_printer, root, output);
 }
 
-void gen_offline_cache_key(IRNode *root, std::string *output) {
+void gen_offline_cache_key(Program *prog, IRNode *root, std::string *output) {
```

Comment:
nit: Why is this implemented inside `ir_printer.cpp`? I think it should be in its own file under `analysis`, WDYT?

Reply:
Agree! But this is temporary. Currently, `gen_offline_cache_key` depends on `IRPrinter` to generate AST-string . Next step, I will create a new class, maybe `ASTOfflineCacheKeyGenerator`, to replace `IRPrinter` to generate AST-Offline-Cache-Key.

But now, because `IRPrinter` is directly defined in ir_printer.cpp without declaring in *.h, I can only put `gen_offline_cache_key` in ir_printer.cpp.

Reply:
@k-ye WDYT about https://github.com/taichi-dev/taichi/pull/4716#discussion_r846707738

Reply:
Sorry for the late reply, sounds good!